### PR TITLE
[cleanup] Remove `Format Notation` command

### LINKED ISSUE #5534: Format Notation needs documentation
Note: the issue was created automatically with bugzilla2github tool

Original bug ID: BZ#5534
From: @psteckler
Reported version: 8.6
CC:   @ejgallego

### PR DESCRIPTION
**Pull request by Alizter**
closes #5534

Seems to have been a half-finished experiment. It is unused and supposedly useless.

### CODE REVIEW

File: vernac/vernacentries.ml [hunk 1]
```diff
@@ -2308,10 +2308,6 @@ let translate_vernac ?loc ~atts v = let open Vernacextend in match v with
     vtdefault(fun () -> with_section_locality ~atts vernac_open_close_scope (b,s))
   | VernacNotation (infix,c,infpl,sc) ->
     vtdefault(fun () -> vernac_notation ~atts ~infix c infpl sc)
-  | VernacNotationAddFormat(n,k,v) ->
-    vtdefault(fun () ->
-        unsupported_attributes atts;
-        Metasyntax.add_notation_extra_printing_rule n k v)
```
Comment (ejgallego) [inline vernac/vernacentries.ml:2314]: This should be cleaned up too, I'll push a commit.

**Comment by Zimmi48** [comment]
Should we deprecate in 8.17 and remove in 8.18? Or should we remove without warning?

**Comment by ejgallego** [comment]
We hope to have something similar in coq-layout-engine but indeed, in this case the rules are engine-specific (keyed by notation + likely other stuff)

**Review by Zimmi48 — APPROVED** [review]
Let's wait for @gares's input.

**Comment by jfehrle** [comment]
"closes #5534" will mark that PR as completed when this PR is merged, which is incorrect.  It should be marked as "close as not planned", which AFAIK has to be done manually.

**Comment by Zimmi48** [comment]
Unfortunately, I am not aware of GitHub providing a way to automatically close as not planned and it seems better to use the auto-closing feature to not forget to close related issues. It is still possible to fix the closed status afterward. We can also consider that we have "resolved" the documentation issue by removing the item to be documented.

**Comment by gares** [comment]
This is indeed a feature no frontend uses, as far as I know.
The need for the feature is clearly there, eg coqdoc and coqtex have/had hacks to do the same (as in writing alternative formatting rules in comments).

I don't have immediate plans to use the feature, which lets one associate some alternative (to text) formatting directive to notations. So I can't oppose to its removal, but I find this way of fixing a documentation issue a bit unorthodox, one could also document it as in "formar bla bla, let one specify alternative formatting rules; as of today only the rules accessible via ML APIs".

Do we really want to remove this thing? Where would layouts for coq-layout-engine be stored, if not where these additional formatting rules are stored?

**Comment by Alizter** [comment]
@gares Since you are hesitant on removing it, I say we document it instead.

**Comment by ejgallego** [comment]
I actually tried to use this quite a long time ago, but indeed my conclusion is that it was too primitive for it to work, so the cleanup I think is worth it, but of course we don't need to remove it. But in the end, I think it is effectively dead code.

`coq-layout-engine` provides to the rendered a structure with the notation key and the arguments, and the rendered will implement what it needs, see https://github.com/jscoq/jscoq/blob/99f18653b5388e2603627ff45d1fb1067a9506fb/coq-layout-engine/boxModel.mli#L69

So the format just implements a map from notation key to string, then the string is implementation dependent, and usually not enough [for latex maybe?] as you in general are going to have state in the rendered with will affect the expression.

Lean overcomes this by using the full programming language as the action.

But even a more pressing problem is that the notation formatting system to which this info is attached doesn't follow a box model, so that's where you hit a dead end IMHO.

[I gave a talk about the layout engine at Irif some time ago where we discussed this btw]

So with all this in mind, and seeing the number of lines cleaned up in an already complex file, I'd say let's remove it, and bring it back if people have a use for it. But if we wanna keep and document it that's OK too.

**Comment by ejgallego** [comment]
Indeed, I just saw the comment here https://github.com/coq/coq/issues/5534#issuecomment-1261450115 , using this requires writing your own pretty printer, so it may as well store the info in its own format and table.

I just remembered that using SerAPI you can query this info, but it is not usable with the current printer. There is this directory `sertex` which is the origin of the layout engine today.

**Comment by Zimmi48** [comment]
Any option is fine but let's not just decide to document it and forget about it. Either we consider this useful and it should be documented in time for the next release, or it is not useful in its current form and it should be removed.

**Comment by ejgallego** [comment]
Yeah let's clean this up.

**Review by SkySkimmer — APPROVED** [review]

**Comment by SkySkimmer** [comment]
@coqbot merge now

**Comment by ejgallego** [comment]
Also coq-serapi was broken but I've pushed a fix upstream, ping me if for some reason the CI is still broken.

**Pull request merged**